File: app/src/main/java/com/nathaniel/motus/cavevin/view/BottomBarFragment.java
package com.nathaniel.motus.cavevin.view;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.ImageView;

import com.nathaniel.motus.cavevin.R;


/**
 * A simple {@link Fragment} subclass.
 */
public class BottomBarFragment extends Fragment implements View.OnClickListener {

    //Buttons declaration
    ImageView mAllWinesButton,mRedButton,mWhiteButton,mPinkButton;

    //Shared preferences tags
    private static final String CURRENT_CELLAR_INDEX="Current cellar index";
    private static final String CURRENT_TYPE_FILTER="Current type filter";
    private static final String CURRENT_SORT_OPTION ="Current sort options";

    //Callback
    onBottomBarClickedListener mCallBack;


    public BottomBarFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View result=inflater.inflate(R.layout.fragment_bottom_bar, container, false);

//        Inflate buttons
        mAllWinesButton=result.findViewById(R.id.bottombar_all_wines_button);
        mRedButton=result.findViewById(R.id.bottombar_red_button);
        mWhiteButton=result.findViewById(R.id.bottombar_white_button);
        mPinkButton=result.findViewById(R.id.bottombar_pink_button);

//        Activate the last chosen
        String currentTypeFilter= getActivity().getPreferences(Context.MODE_PRIVATE).getString(CURRENT_TYPE_FILTER,"all");
        switch (currentTypeFilter){
            case "red":
                mRedButton.setBackgroundColor(result.getContext().getResources().getColor(R.color.colorAccent));
                break;
            case "white":
                mWhiteButton.setBackgroundColor(result.getContext().getResources().getColor(R.color.colorAccent));
                break;
            case "pink":
                mPinkButton.setBackgroundColor(result.getContext().getResources().getColor(R.color.colorAccent));
                break;
            default:
                mAllWinesButton.setBackgroundColor(result.getContext().getResources().getColor(R.color.colorAccent));
        }

//        Put on clicklistener
        mAllWinesButton.setOnClickListener(this);
        mRedButton.setOnClickListener(this);
        mWhiteButton.setOnClickListener(this);
        mPinkButton.setOnClickListener(this);

//        create callback
        createCallBackToParentActivity();

        return result;
    }

    @Override
    public void onClick(View v) {
        String buttonTag=v.getTag().toString();
        //change the colors
        mAllWinesButton.setBackgroundColor(v.getContext().getResources().getColor(R.color.colorPrimary));
        mRedButton.setBackgroundColor(v.getContext().getResources().getColor(R.color.colorPrimary));
        mWhiteButton.setBackgroundColor(v.getContext().getResources().getColor(R.color.colorPrimary));
        mPinkButton.setBackgroundColor(v.getContext().getResources().getColor(R.color.colorPrimary));
        v.setBackgroundColor(v.getContext().getResources().getColor(R.color.colorAccent));

        mCallBack.onBottomBarItemClicked(v,buttonTag);
    }

//    **********************************************************************************************
//    Callback subs
//    **********************************************************************************************

    public interface onBottomBarClickedListener{
        public void onBottomBarItemClicked(View v, String buttonTag);
    }

    private void createCallBackToParentActivity(){
        try {
            mCallBack=(onBottomBarClickedListener)getActivity();
        }catch (ClassCastException e){
            throw new ClassCastException(e.toString()+" must implement onItemClickedListener");
        }
    }

}

File: app/src/main/java/com/nathaniel/motus/cavevin/view/BottomBarFragment.kt
package com.nathaniel.motus.cavevin.view

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.fragment.app.Fragment
import com.nathaniel.motus.cavevin.databinding.FragmentBottomBarBinding

class BottomBarFragment(private val fragment: BottomBarParentFragment) : Fragment() {

    private var _binding: FragmentBottomBarBinding? = null
    private val binding get() = _binding
    private val buttonClickListener=View.OnClickListener{
        fragment.notifyButtonClicked(it.tag as String)
    }
    var stillButtonState=true
    var sparklingButtonState=true
    var redButtonState=true
    var whiteButtonState=true
    var pinkButtonState=true
    var emptyButtonState=true

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val fragmentBinding=FragmentBottomBarBinding.inflate(inflater,container,false)
        _binding=fragmentBinding
        return fragmentBinding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setButtonTags()
        applyButtonStates()
    }

    private fun setButtonTags(){
        binding?.bottomBarStillButton?.tag= STILL_BUTTON
        binding?.bottomBarStillButton?.setOnClickListener(buttonClickListener)
        binding?.bottomBarSparklingButton?.tag= SPARKLING_BUTTON
        binding?.bottomBarSparklingButton?.setOnClickListener(buttonClickListener)
        binding?.bottomBarRedButton?.tag= RED_BUTTON
        binding?.bottomBarRedButton?.setOnClickListener(buttonClickListener)
        binding?.bottomBarWhiteButton?.tag= WHITE_BUTTON
        binding?.bottomBarWhiteButton?.setOnClickListener(buttonClickListener)
        binding?.bottomBarPinkButton?.tag= PINK_BUTTON
        binding?.bottomBarPinkButton?.setOnClickListener(buttonClickListener)
        binding?.bottomBarEmptyButton?.tag= EMPTY_BUTTON
        binding?.bottomBarEmptyButton?.setOnClickListener(buttonClickListener)
    }

    fun applyButtonStates(){
        applyButtonState(binding?.bottomBarEmptyButton,emptyButtonState)
        applyButtonState(binding?.bottomBarRedButton,redButtonState)
        applyButtonState(binding?.bottomBarPinkButton,pinkButtonState)
        applyButtonState(binding?.bottomBarSparklingButton,sparklingButtonState)
        applyButtonState(binding?.bottomBarStillButton,stillButtonState)
        applyButtonState(binding?.bottomBarWhiteButton,whiteButtonState)
    }

    private fun applyButtonState(imageView: ImageView?, state:Boolean){
        if (state)
            imageView?.alpha=1f
        else imageView?.alpha=0.1f
    }

    interface BottomBarParentFragment{
        fun notifyButtonClicked(buttonTag:String)
    }

    companion object{
        const val STILL_BUTTON="stillButton"
        const val SPARKLING_BUTTON="sparklingButton"
        const val RED_BUTTON="redButton"
        const val WHITE_BUTTON="whiteButton"
        const val PINK_BUTTON="pinkButton"
        const val EMPTY_BUTTON="emptyButton"
    }
}